### PR TITLE
Drop support for Python 2.7, 3.5, 3.6.

### PR DESCRIPTION
**Pull request by icemac**

### CODE REVIEW

**Review by dataflake — APPROVED** [review]

File: CHANGES.rst [hunk 1]
```diff
@@ -1,9 +1,13 @@
 Changelog
 =========
 
-3.7 (unreleased)
+4.0 (unreleased)
 ----------------
 
+- Add support for Python 3.11.
```
Comment (d-maurer) [inline CHANGES.rst:7]: `CHANGES.rst` says `Add support for Python 3.11` but in the diff I have not seen 3.11 additions (may have been already there).

File: src/Persistence/mapping.py [hunk 2]
```diff
@@ -35,55 +33,34 @@ def __new__(cls, *args, **kw):
...
-        __metaclass__ = _Meta
-
-        def __setstate__(self, state):
-            if 'data' not in state:
-                state['data'] = state['_container']
-                del state['_container']
-            self.__dict__.update(state)
-
-else:  # pragma: PY3
+def with_metaclass(meta, *bases):
```
Comment (d-maurer) [inline src/Persistence/mapping.py:36]: Could this not be simplified?
The former complexity came from Python 2 and 3 using different mechanisms to specify a metaclass.

**Review by d-maurer — APPROVED** [review]

File: CHANGES.rst (hunk 1 above)
Comment (dataflake) [reply CHANGES.rst]: Python 3.11 support was already there, I removed the entry.

File: src/Persistence/mapping.py (hunk 2 above)
Comment (icemac) [reply src/Persistence/mapping.py:36]: I looked at the code and it seemed to complex for an easy refactoring. Any help is welcome.

Comment (d-maurer) [reply src/Persistence/mapping.py]: I push a simplification trial: `type()` and `._bases__` give the same result.

Comment (icemac) [reply src/Persistence/mapping.py]: Thank you!

**Comment by icemac** [comment]
Thank you for reviewing this PR. 😃

**Pull request merged**